src/helpers/fiware-token.helper.ts

```diff
@@ -0,0 +1,46 @@
+import { CACHE_MANAGER, HttpService, Inject, Injectable, Logger } from "@nestjs/common"
+import { Cache } from 'cache-manager'
+import { FiwareDataTargetConfiguration } from "../entities/interfaces/fiware-data-target-configuration.interface";
+
+type TokenEndpointResponse = {
+    access_token: string,
+    expires_in: number,
+}
+
+@Injectable()
+export class AuthenticationTokenProvider {
+
+    private readonly logger = new Logger(AuthenticationTokenProvider.name);
+
+    constructor(private httpService: HttpService, @Inject(CACHE_MANAGER) private cacheManager: Cache) {
+    }
+
+    async getToken(config: FiwareDataTargetConfiguration): Promise<string> {
+        const key = config.clientId + config.clientSecret;
+        const token = await this.cacheManager.get<string>(key)
+        if (token) {
+            this.logger.debug('Token found')
+            return token
+        } else {
+            try {
+                const encodedCredentials = Buffer.from(`${config.clientId}:${config.clientSecret}`).toString('base64');
+                const params = new URLSearchParams([['grant_type', 'client_credentials']])
+                const { data }: { data: TokenEndpointResponse } = await this.httpService.post(config.tokenEndpoint, params, {
+                    headers: {
+                        'Authorization': `Basic ${encodedCredentials}`,
```

Comment:
Why is client id and secret passed as a base64 encoded string in the Authorization header instead of a URL parameter (like with grant_type)? 
For reference: https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/

Reply:
I think it was based on some sample, but I have changed to use to standard way as described in your link